lib/shoryuken/middleware/server/timing.rb

```diff
@@ -4,17 +4,20 @@ module Server
       class Timing
         include Util
 
-        def call(_worker, queue, _sqs_msg, _body)
+        def call(_worker, _queue, sqs_msg, _body)
           started_at = Time.now
 
           logger.info { "started at #{started_at}" }
 
           yield
 
-          total_time = elapsed(started_at)
+          ended_at = Time.now
+          total_time = (ended_at - started_at) * 1000
 
-          if (total_time / 1000.0) > (timeout = Shoryuken::Client.queues(queue).visibility_timeout)
-            logger.warn { "exceeded the queue visibility timeout by #{total_time - (timeout * 1000)} ms" }
+          if ended_at > sqs_msg.become_available_at
```

Comment:
@moyuanhuang isn't this still using the queue visibility instead of the message specific visibility timeout? 

This `Shoryuken::Client.queues(queue).visibility_timeout` will give the queue visibility, not the visibility for the specific message.

Reply:
The `sqs_msg.become_available_at` is initialized using that value, and thus will remain the same with the queue level visibility timeout until `sqs_msg.visibility_timeout(timeout)` is called. Within that method call, the `become_available_at` will be set to `Time.now + timeout`. ([code])

[code]: https://github.com/phstc/shoryuken/pull/592/files/57c6640f2ff7e03b0a69ea25aaaebd59b13904fe#diff-eb88953e947b8431ea88b452a873eac8R36

Reply:
@phstc wanna take another look?

Reply:
Sure! Sorry for taking longer on this, I will 👀 over the weekend. But makes sense.

I missed the setting part in the code diff, thanks for pointing that out.

Reply:
If we go with the suggested change of only introducing a getter `Message#visibility_timeout` the only diff will be:

```ruby
# from this
# (timeout = Shoryuken::Client.queues(queue).visibility_timeout)
# to

(timeout = sqs_msg.visibility_timeout)
```

Right?